src/overlays/actors/ovl_En_Encount1/z_en_encount1.c

```diff
@@ -19,13 +26,313 @@ const ActorInit En_Encount1_InitVars = {
+    Player* player = PLAYER;
+    s32 bgId;
+    CollisionPoly* floorPoly;
+    Vec3f spawnPos;
+    f32 floorY;
+
+    if (this->timer == 0) {
+        this->timer = 10;
+        if ((fabsf(player->actor.posRot.pos.y - this->actor.posRot.pos.y) > 100.0f) ||
+            (this->actor.xzDistFromLink > this->spawnRange)) {
+            this->outOfRangeTimer++;
+        } else {
+            this->outOfRangeTimer = 0;
+            if ((this->curNumSpawn < this->maxCurSpawns) && (this->totalNumSpawn < this->maxTotalSpawns)) {
+                spawnPos.x = this->actor.posRot.pos.x + Math_Rand_CenteredFloat(50.0f);
+                spawnPos.y = this->actor.posRot.pos.y + 120.0f;
+                spawnPos.z = this->actor.posRot.pos.z + Math_Rand_CenteredFloat(50.0f);
+                floorY = func_8003C9A4(&globalCtx->colCtx, &floorPoly, &bgId, &this->actor, &spawnPos);
+                if (floorY <= -32000.0f) {
+                    return;
+                }
+                spawnPos.y = floorY;
+                if (Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_EN_TITE, spawnPos.x,
+                                       spawnPos.y, spawnPos.z, 0, 0, 0, -1) != NULL) { // Red tektite
+                    this->curNumSpawn++;
+                    this->totalNumSpawn++;
+                } else {
+                    // Cannot spawn!
+                    osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 発生できません！ ☆☆☆☆☆\n" VT_RST);
+                    osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 発生できません！ ☆☆☆☆☆\n" VT_RST);
+                    osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 発生できません！ ☆☆☆☆☆\n" VT_RST);
+                }
+            }
+        }
+    }
+}
+
+void EnEncount1_SpawnStalchildOrWolfos(EnEncount1* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    f32 spawnDist;
+    s16 spawnAngle;
+    s16 spawnId;
+    s16 spawnParams;
+    s16 kcOver10;
+    s16 tempmod;
+    Vec3f spawnPos;
+    CollisionPoly* floorPoly;
+    s32 bgId;
+    f32 floorY;
+
+    if (globalCtx->sceneNum != SCENE_SPOT00) {
+        if ((fabsf(player->actor.posRot.pos.y - this->actor.posRot.pos.y) > 100.0f) ||
+            (this->actor.xzDistFromLink > this->spawnRange)) {
+            this->outOfRangeTimer++;
+            return;
+        }
+    } else if (!gSaveContext.nightFlag || (Player_GetMask(globalCtx) == PLAYER_MASK_BUNNY)) {
+        this->killCount = 0;
+        return;
+    }
+
+    this->outOfRangeTimer = 0;
+    spawnPos = this->actor.posRot.pos;
+    if ((this->curNumSpawn < this->maxCurSpawns) && (this->totalNumSpawn < this->maxTotalSpawns)) {
+        while ((this->curNumSpawn < this->maxCurSpawns) && (this->totalNumSpawn < this->maxTotalSpawns)) {
+            if (globalCtx->sceneNum == SCENE_SPOT00) {
+                if ((player->unk_89E == 0) || (player->actor.floorPolySource != 0x32) ||
+                    !(player->actor.bgCheckFlags & 1) || (player->stateFlags1 & 0x08000000)) {
```

Comment:
state flags check can be & (1<<27)

Reply:
our flags are in hex for now, i dont think it matters too much until they become defines later anyway

Reply:
I don't think we've been doing this in the past. Not opposed to it in principle (though eventually they'll be macros) but if so there's a *lot* more to change.